### ISSUE TITLE
[P2P] V1 P2P First Iteration - Followups

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket#61]
# Objective
Address all of the discussion and implementation points left in the primary PR: https://github.com/pokt-network/pocket/pull/46

# Origin Document
[Issue #24: V1 P2P Module - First Iteration](https://github.com/pokt-network/pocket/issues/25)

# Goals
## Deliverables
- [x] Address all outstanding TODOs in the P2P module
- [ ] Address all of the open items below

## Testing Methodology

```
$ go mod vendor && go mod tidy
$ make protogen_clean && make protogen_local
$ make mockgen
$ make test_all
```

## Non-goals
- Add any new functionality to the P2P module

---

### Short-term refactor
- [x] Consider creating a helper for `s.closed <- struct{}{}` called `SignalClose` so it's more semantically explicit as one reads the codebase

### Mid-term refactor
- [ ] Sync with @Olshansk on how we can use mockgen for socket_mock.go (see utility in consensus PR as a reference)
- [ ] Add additional non-happy path test cases

### Large refactor
- [ ] Design & consolidate `logger` with the rest of the repository
- [ ] Replace `WireCodec` with `protobufs`
- [x] #66

### General tasks
- [x] #67

## Discuss
- [x] `The write and read routines will both exit for the same reasons, so having just one of them block is sufficient` . IMO we should await on both (see [this](https://dave.cheney.net/2013/04/30/curious-channels) as a reference). If we don't, and simply assume they both exit for the same reason, it looks like an opportunity for a bug if one breaks but the other doesn't.

**Comment by derrandz** [comment]
Most of what's addressable at this point of the PR has been addressed. The remaining are as highlighted above:

- Refactors to consolidate the logger functionality with the rest of the repository
- Replacing WireCodec with protobufs or flatbuffers.
- Using Mockgen

**Comment by derrandz** [comment]
Here is a non-exhaustive list of questions to be answered for the logging topic (_by @Olshansk_)

1. Do we build our own logging library?
1.5 If so, how do we design / use it?
2. Do we use another open source logging library?
3. What logging libraries do other projects use?

No urgency on the topic, will probably be attended to in v1 next development cycle

**Comment by Olshansk** [comment]
@derrandz Can you let me know if this task is still applicable?

**Comment by Olshansk** [comment]
Actually, we could use this to track the in  https://github.com/pokt-network/pocket/pull/80 + reviews. Wdyt?

**Comment by derrandz** [comment]
#67 and #66 are no longer applicable. The other two are though.

I think we can execute #80 separately because our of the two remaining tasks, the logging one will be acted upon in the telemetry milestone.

For the Codecs task, it is blocked by the IO tasks (#86), and we will get to it by the time we start worrying about #70

Wdyt?

**Comment by Olshansk** [comment]
So, course of action:
1. Close https://github.com/pokt-network/pocket/issues/66
2. Close https://github.com/pokt-network/pocket/issues/6y (which you've already done)
3. I create an issue to track https://github.com/pokt-network/pocket/pull/80
4. Keep https://github.com/pokt-network/pocket/issues/86 and https://github.com/pokt-network/pocket/issues/70 for future work

**Comment by Olshansk** [comment]
Closing this since we are moving forward with Pre2P and some of the other tasks cover followup work.
